lib/reporters/Reporter.js

```diff
@@ -1,19 +1,26 @@
+/** @typedef {import("webpack/lib/Compiler")} Compiler */
+/**
+ * @typedef {Object} ReporterOptions
+ * @property {Compiler} ReporterOptions.compiler webpack compiler instance
+ * @property {Object} ReporterOptions.config webpack configuration object
+ */
+
 module.exports = class Reporter {
   /**
    * @constructor
-   * @param compiler  {Object}  A webpack Compiler instance https://webpack.js.org/api/node/#compiler-instance
+   * @abstract
+   * @param {ReporterOptions} options
    */
-  // eslint-disable-next-line no-unused-vars
   constructor(options) {
     this.compiler = options.compiler;
     this.config = options.config;
   }
 
   /**
    * @method render
-   * @param error  {Error}   An Error object
-   * @param stats  {Object}  A webpack stats object https://webpack.js.org/api/node/#stats-object
+   * @param {Error} error An Error object
+   * @param {Object} stats A webpack stats object https://webpack.js.org/api/node/#stats-object
    */
-  // eslint-disable-next-line no-unused-vars
-  render(error, stats) {}
+
+  render(error, stats) {} // eslint-disable-line no-unused-vars
```

Comment:
the ESLint rule states that comments should be on the line above, but it lets you get away with it for eslint comments. that's whack. revert this please.